diff --git a/Makefile b/Makefile
index 28efb6a..89d1282 100644
--- a/Makefile
+++ b/Makefile
@@ -298,10 +298,8 @@ compilers_engines := $(strip $(compilers_engines))
 #####
 
 build_wasmer_extra_features := wasm-c-api
-ifneq ($(IS_WINDOWS), 1)
-	ifneq (,$(filter 1 true,$(ENABLE_NAPI_V8)))
-		build_wasmer_extra_features += napi-v8
-	endif
+ifneq (,$(filter 1 true,$(ENABLE_NAPI_V8)))
+	build_wasmer_extra_features += napi-v8
 endif
 
 # Small trick to define a space and a comma.
